### ISSUE TITLE
TypeError: Cannot read property 'insertBefore' of null

### ISSUE THREAD

**Issue by ittayd** [angular/angular.js#13174]
Using Angular 1.4.7
```
TypeError: Cannot read property 'insertBefore' of null
    at forEach.after (https://ajax.googleapis.com/ajax/libs/angularjs/1.4.7/angular.js:3453:13)
    at Object.JQLite.(anonymous function) [as after] (https://ajax.googleapis.com/ajax/libs/angularjs/1.4.7/angular.js:3537:17)
    at domInsert (https://ajax.googleapis.com/ajax/libs/angularjs/1.4.7/angular.js:5050:35)
    at Object.$provide.$get.$$animateQueue.enter (https://ajax.googleapis.com/ajax/libs/angularjs/1.4.7/angular.js:5213:9)
    at https://ajax.googleapis.com/ajax/libs/angularjs/1.4.7/angular.js:24071:26
    at publicLinkFn (https://ajax.googleapis.com/ajax/libs/angularjs/1.4.7/angular.js:7554:29)
    at $$sanitizeUriProvider.$get.boundTranscludeFn (https://ajax.googleapis.com/ajax/libs/angularjs/1.4.7/angular.js:7699:16)
    at controllersBoundTransclude (https://ajax.googleapis.com/ajax/libs/angularjs/1.4.7/angular.js:8316:18)
    at Object.ngIfWatchAction [as fn] (https://ajax.googleapis.com/ajax/libs/angularjs/1.4.7/angular.js:24062:15)
    at Scope.$get.Scope.$digest (https://ajax.googleapis.com/ajax/libs/angularjs/1.4.7/angular.js:15826:29)
```

Trying to analyze, it looks like ngIf adds a comment and doesn't insert it into the dom before calling $animage.enter which calls domInsert that then assumes the comment has a parent (it doesn't)

Obviously, ngIf works most of the time. What I can add is that I'm using ux-datagrid and this failure is on the first line it tries to draw that has the ngIf

**Comment by Narretz** [comment]
Thanks for the report. Can you create a minimal demo that shows the problem, preferrably without any third-party modules? Otherwise it's difficult to act on.

**Comment by ittayd** [comment]
Probably not worth your time, so I'll close it. If you do want to investigate, it happens when you have ng-if on the root element in a template of ux-datagrid

**Comment by Narretz** [comment]
No need to close it so fast. We just need a repro that shows the problem is not part of the another module. Maybe ux-datagrid is using the angular APIs in a wrong way, for example.

**Comment by ittayd** [comment]
https://jsfiddle.net/nwx6vgrm/2/

**Comment by Narretz** [comment]
The fiddle didn't work for me (module missing somehow), so I put it in a plnkr. The error is a bit different, but I assume it's the same problem: http://plnkr.co/edit/VgMorP1pMNWIHhlOaQgG?p=preview
Definitely needs some investigation. It's possible that the ux-grid is doing something that's throwing the core off, though, so you might want to open an issue there too.

**Comment by Narretz** [comment]
It's probably because ng-if is on the first element of the template. If you wrap it with another `<div>`, it works.

**Comment by SamuelMS** [comment]
@Narretz I can confirm wrapping the ng-if in an extra div resolved the issue for me as well. It's worth noting that my use case was a bit peculiar, but didn't rely on any external libraries (like ux-grid) -- just pure Angular code.

I can probably throw together a fiddle/pen/plunk if that'd help.

**Comment by Narretz** [comment]
@SamuelMS Runnable examples, are always appreciated!

**Comment by SamuelMS** [comment]
@Narretz Sure thing. I threw together a quick fiddle: https://jsfiddle.net/s8g2nroy

It's a contrived example, but triggers the error. I mentioned several ways to solve the issue for this particular example in the comments, but none of them are really a general-purpose solution.

I think this just boils down to the directive priority, with ng-repeat at 1000 and ng-if at 600. When the $digest cycle fires, it tries to update ng-repeat with the modified scope value of targetString, but the element has already been emptied -- and only once that finishes would it try to remove the element via ng-if.

It's interesting that wrapping the element with a container resolves the issue.

**Comment by Narretz** [comment]
@SamuelMS Thanks for the demo. The original issue had a different error, though. So it looks like your problem is different. Could you please open a new issue with description and demo? That way it's easier to track.

I'm closing this issue as we don't have a reproduction.

**Comment by SamuelMS** [comment]
@Narretz Really? They're both related to ng-if and the error message is the following in both as well:

```TypeError: Cannot read property 'insertBefore' of null```

**Comment by Narretz** [comment]
Yes, but we don't have a demo for the reported problem. Or did I miss it? And @ittayd said it had something to do with animate. In any case, this is a bit of a mess now, and I just want anyone who opens an issue to be able to see what's going on by reading the first post (ideally), not searching for demos etc.
